gcloud-java-pubsub/src/main/java/com/google/gcloud/pubsub/spi/SubscriberApi.java

```diff
@@ -0,0 +1,663 @@
+
+
+// AUTO-GENERATED DOCUMENTATION AND SERVICE - see instructions at the top of the file for editing.
+/**
+ * The service that an application uses to manipulate subscriptions and to
+ * consume messages from a subscription via the Pull method.
+ */
+@javax.annotation.Generated("by the veneer generator")
+public class SubscriberApi implements AutoCloseable {
+
+  // =========
+  // Constants
+  // =========
+
+  /**
+   * The default address of the service.
+   */
+  public static final String SERVICE_ADDRESS = "pubsub-experimental.googleapis.com";
+
+  /**
+   * The default port of the service.
+   */
+  public static final int DEFAULT_SERVICE_PORT = 443;
+
+
+  public static final ApiCallable<Subscription, Subscription>
+      CREATE_SUBSCRIPTION = ApiCallable.create(SubscriberGrpc.METHOD_CREATE_SUBSCRIPTION);
+  public static final ApiCallable<GetSubscriptionRequest, Subscription>
+      GET_SUBSCRIPTION = ApiCallable.create(SubscriberGrpc.METHOD_GET_SUBSCRIPTION);
+  public static final ApiCallable<ListSubscriptionsRequest, ListSubscriptionsResponse>
+      LIST_SUBSCRIPTIONS = ApiCallable.create(SubscriberGrpc.METHOD_LIST_SUBSCRIPTIONS);
+  public static final ApiCallable<DeleteSubscriptionRequest, Empty>
+      DELETE_SUBSCRIPTION = ApiCallable.create(SubscriberGrpc.METHOD_DELETE_SUBSCRIPTION);
+  public static final ApiCallable<ModifyAckDeadlineRequest, Empty>
+      MODIFY_ACK_DEADLINE = ApiCallable.create(SubscriberGrpc.METHOD_MODIFY_ACK_DEADLINE);
+  public static final ApiCallable<AcknowledgeRequest, Empty>
+      ACKNOWLEDGE = ApiCallable.create(SubscriberGrpc.METHOD_ACKNOWLEDGE);
+  public static final ApiCallable<PullRequest, PullResponse>
+      PULL = ApiCallable.create(SubscriberGrpc.METHOD_PULL);
+  public static final ApiCallable<ModifyPushConfigRequest, Empty>
+      MODIFY_PUSH_CONFIG = ApiCallable.create(SubscriberGrpc.METHOD_MODIFY_PUSH_CONFIG);
+
+
+
+  private static PageDescriptor<ListSubscriptionsRequest, ListSubscriptionsResponse, Subscription> LIST_SUBSCRIPTIONS_PAGE_DESC =
+      new PageDescriptor<ListSubscriptionsRequest, ListSubscriptionsResponse, Subscription>() {
+        @Override public Object emptyToken() {
+          return "";
+        }
+        @Override public ListSubscriptionsRequest injectToken(
+            ListSubscriptionsRequest payload, Object token) {
+          return ListSubscriptionsRequest
+            .newBuilder(payload)
+            .setPageToken((String) token)
+            .build();
+        }
+        @Override
+        public Object extractNextToken(ListSubscriptionsResponse payload) {
+          return payload.getNextPageToken();
+        }
+        @Override
+        public Iterable<Subscription> extractResources(ListSubscriptionsResponse payload) {
+          return payload.getSubscriptionsList();
+        }
+      };
+
+
+
+
+
+  private static String ALL_SCOPES[] = {
+    "https://www.googleapis.com/auth/pubsub"
+  };
+
+  public static final PathTemplate PROJECT_PATH_TEMPLATE =
```

Comment:
> It make sense if you need them across the service classes (XXXApi).

So the story isn't so clean. There are indeed paths that are the same across the service api classes, but currently we don't do anything to match them up, so you'll see them repeated in each service api class (see for example 'createProject'). It is set up this way because there is currently no single point where these are defined; in our code generation, we currently assume that references to the same pattern across multiple api method calls are the same thing, but only within the scope of a single service definition. It might be more ideal to merge the identifiers together across multiple service definitions. This would have the advantage of being better typed, and there would be less "apparent" duplication in the wrapper layer. It has the slight disadvantage though of making them more distant from the api calls that would need them, making them slightly less discoverable.